WordPress/src/main/java/org/wordpress/android/ui/posts/adapters/PostListAdapter.kt

```diff
@@ -186,75 +170,89 @@ class PostListAdapter(
 
     override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
         // nothing to do if this is the static endlist indicator
-        if (getItemViewType(position) == VIEW_TYPE_ENDLIST_INDICATOR) {
+        val viewType = getItemViewType(position)
+        if (viewType == VIEW_TYPE_ENDLIST_INDICATOR) {
+            return
+        }
+        if (viewType == VIEW_TYPE_LOADING) {
             return
         }
 
-        val post = posts[position]
-        val context = holder.itemView.context
-
-        if (holder is PostViewHolder) {
-            if (StringUtils.isNotEmpty(post.title)) {
-                // Unescape HTML
-                val cleanPostTitle = StringEscapeUtils.unescapeHtml4(post.title)
-                holder.title.text = cleanPostTitle
-            } else {
-                holder.title.text = context.resources.getText(R.string.untitled_in_parentheses)
-            }
+        listManager?.getItem(position)?.let { post ->
+            val context = holder.itemView.context
 
-            var cleanPostExcerpt = PostUtils.getPostListExcerptFromPost(post)
+            if (holder is PostViewHolder) {
```

Comment:
AFAIU the holder should always be an instance of PostViewHolder. We may consider throwing an exception in the else branch or remove the if check so it crashes asap.

Reply:
Agreed, fixed in 8719cdf4fd7a263d114ee14e44f6402ffb8fec85.